tools/pd-simulator/simulator/cases/cases.go

```diff
@@ -110,3 +111,45 @@ func NewCase(name string) *Case {
 	}
 	return nil
 }
+
+func leaderAndRegionIsUniform(leaderCount, regionCount, regionNum int, threshold float64) bool {
+	return leaderIsUniform(leaderCount, regionNum, threshold) && regionIsUniform(regionCount, regionNum, threshold)
+}
+
+func leaderIsUniform(leaderCount, regionNum int, threshold float64) bool {
+	return isUniform(leaderCount, regionNum/3, threshold)
+}
+
+func regionIsUniform(regionCount, regionNum int, threshold float64) bool {
+	return isUniform(regionCount, regionNum, threshold)
+}
+
+func isUniform(count, meanCount int, threshold float64) bool {
+	maxCount := int((1.0 + threshold) * float64(meanCount))
+	minCount := int((1.0 - threshold) * float64(meanCount))
+	return minCount <= count && count <= maxCount
+}
+
+func getStoreNum() int {
+	storeNum := simutil.CaseConfigure.StoreNum
+	if storeNum < 3 {
+		simutil.Logger.Fatal("Store num should be larger than 3.")
+	}
+	return storeNum
+}
+
+func getRegionNum() int {
+	regionNum := simutil.CaseConfigure.RegionNum
+	if regionNum <= 0 {
+		simutil.Logger.Fatal("Region num should be larger than 0.")
+	}
+	return regionNum
+}
+
+func getFullStoreNum(storeNum int, fullRatio float64) uint64 {
```

Comment:
I'm still confused about the full store here means

Reply:
how about `noEmptyNum`?